### ISSUE TITLE
Unable to create or import a library on Windows: out of memory-Error opening database

### ISSUE THREAD

**Issue by gomera** [YACReader/yacreader#37]
I have my current library on Linux and use Paragon ExtFS for mounting all my drivers into windows without any problems for years. The issue is that I cannot import that folder nor even create a new empty one on those partitions because of this weird error:

![image](https://user-images.githubusercontent.com/1063134/52513673-25067580-2beb-11e9-8cbe-ba66bfbe72db.png)

When I restart the app, the apps says that the library was created with an older version and has to be recreated. The metafile within the library folders is completely empty.

The logs shows nothing, just an empty line with the error I am already seeing on the app.

```
ERROR 2019-02-08T21:52:25.657 Unable to create data base "out of memory-Error opening database" 
```

I cannot use the app at all cause all my partitions are ext4, except the windows and a small one that I am already using for Steam.

If need any more info, please let me know and I'll update this ticket.

**Comment by selmf** [comment]
The ext4 filesystem itself shouldn't be a problem. We have lots of Linux users, including me, and ext4 is the most common filesystem there.

As your setup is special I can't replicate that issue, so I need to take an educated guess. YACReaderLibrary uses sqlite for database management, which in turn uses a file system mechanism called "file locking". This means that the app working on the database locks the file via file system, so no other apps can access it while the database is being processed.
Sqlite is quite picky with this. If the file locking mechanism's implementation is incomplete or faulty, it will not work. This frequently happens for users having their comic collection on a NAS storage, and in their case I either recommend to disable file locking (which has its own drawbacks) or hosting the library on a local partition (on Linux this can be done using symlinks, but it is tricky).

My recommendation in your case:
Try to narrow down the problem as much as possible, and contact Paragon regarding the file locking mechanism. If this is a driver problem, there is not much I can do.

You could also upload a copy of the botched up database (it is located in the hidden .yacreaderlibrary directory in the root of your comics collection) so we can have a look how far the creation actually got.

**Comment by gomera** [comment]
@selmf There is no db after the error, only an empty ".yacreaderlibrary" file, not a folder. Don't know if this is related with sqlite, could be.  I trust the driver for file management, I have been using it for years without any issues. I think clementine uses sqlite, the library it also on a ext4 partition and don't see any issue so far. Don't know if I can help with anything, I know it is hard to fix something you cannot reproduce. Maybe I could setup the dev environment and make some test, just let me know what can I do to help. Thanks !

**Comment by selmf** [comment]
Try deleting the .yacreaderlibrary file and manually create a directory with the same name in its place.

**Comment by gomera** [comment]
Just tried that, I create a folder with a comic and create manually that folder and the importer worked without any issues. So it seems a Paragon issue not been able to create a folder with a dot at the beginning, cause I got an error if I try with File Explorer but with the terminal there was not problem. I've just created them a support request ticket and keep you informed. TBH, don't know why YacReader Library does not work if a I could create the folder manually within the terminal. I think that's another different topic :)
